### ISSUE TITLE
`toObject` produces object does not conform to AsObject

### ISSUE THREAD

**Issue by MrGWest** [improbable-eng/ts-protoc-gen#182]
**Versions of relevant software used**
google-protobuf: 3.8.0
ts-protoc-gen: 0.10.0

**What happened**
When inspecting an object that was returned by the method `toObject` the properties all start with lower case letters. The `.AsObject` type definitions all start with upper case letters.

**What you expected to happen**
Expected the object properties to conform to the `AsObject` type definition.

**How to reproduce it (as minimally and precisely as possible)**:
`var obj = Dog.toObject()`
Step over the above statement and inspect the resulting `obj` object.
Compare this to the definition of `Dog.AsObject`.

**Comment by jonny-improbable** [comment]
Hi @MrGWest, thanks for raising.

I am unable to produce this issue, given the following input proto:

```
syntax = "proto3";

package examplecom;

message CasingMessage {
    string TitleCase = 1;
    string camelCase = 2;
    string snake_case = 3;
}
```

The following artefacts are generated using `protoc` v3.8.0:

*casing_pb.d.ts*
```typescript
export namespace CasingMessage {
  export type AsObject = {
    titlecase: string,
    camelcase: string,
    snakeCase: string,
  }
}
```

*casing_pb.js*
```javascript
proto.examplecom.CasingMessage.toObject = function(includeInstance, msg) {
  var f, obj = {
    titlecase: jspb.Message.getFieldWithDefault(msg, 1, ""),
    camelcase: jspb.Message.getFieldWithDefault(msg, 2, ""),
    snakeCase: jspb.Message.getFieldWithDefault(msg, 3, "")
  };
```

Please could you provide a complete repro for this issue for me to debug? 

Thanks.

**Comment by MrGWest** [comment]
Thank you, @jonny-improbable for the quick response and my apologies for the poor description.

The problem is with field names that (do not follow strict guide lines and) start with a leading underscore.

_exception.proto_
```
message Exception {
    optional string _uuid = 1;
    optional string _created_at = 2;
    optional string _updated_at = 3;
    optional string _deleted_at = 4;
}
```
generates the following
_exception_pb.js_
```
proto.exception.Exception.toObject = function(includeInstance, msg) {
  var f, obj = {
    uuid: (f = jspb.Message.getField(msg, 1)) == null ? undefined : f,
    createdAt: (f = jspb.Message.getField(msg, 2)) == null ? undefined : f,
    updatedAt: (f = jspb.Message.getField(msg, 3)) == null ? undefined : f,
    deletedAt: (f = jspb.Message.getField(msg, 4)) == null ? undefined : f,
}
...
```
_exception_pb.d.ts_
```
export namespace Exception {
  export type AsObject = {
    Uuid?: string,
    CreatedAt?: string,
    UpdatedAt?: string,
    DeletedAt?: string,
  }
}
```

**Comment by jonny-improbable** [comment]
Perfect, thanks for the detailed repro. I was able to reproduce this locally and push, what I hope is a fix in #183.

**Comment by MrGWest** [comment]
Thank you for the quick resolution!
